src/axolotl/utils/models.py

```diff
@@ -352,6 +359,16 @@ def load_model(
             model, use_gradient_checkpointing=cfg.gradient_checkpointing
         )
 
+        # LlamaRMSNorm layers are in fp32 after kbit_training, so we need to
+        # convert them back to fp16/bf16 for flash-attn compatibility.
+        if cfg.flash_attention and cfg.is_llama_derived_model:
+            for name, module in model.named_modules():
```

Comment:
Is this supposed to be before or after load_adapter?

Reply:
this is where it was before we moved it around. The adapter doesn't actually touch any of these, it just loads a new LoraModel w LoraLayers and freezes these.